databroker/_core.py

```diff
@@ -1657,7 +1657,7 @@ def get_table(self,
 
                 # get the first descriptor for this event stream
                 desc = next((d for d in h.descriptors
-                             if d['name'] == stream_name),
+                             if d.get('name') == stream_name),
```

Comment:
just a little devil's advocate, what if somehow `stream_name = None` was passed? Shouldn't be possible, but might be safer
```py3
if d.get('name') == stream_name and d.get("name") is not None
```

Reply:
Stream names, if present, must be strings.